### PR TITLE
refactor: Remove exchangeUp/down,and add EventSource,Optimize API

### PR DESCRIPTION
**Pull request by leonside**
1. Remove ExchangeEvent Indicates the type of the exchangeUp/exchangeDown event
2. Add EventSource and implementation
3. Adjust the API interface

### CODE REVIEW

File: core/context/src/main/java/com/milesight/beaveriot/context/util/ExchangeContextHelper.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            } else if (flowId != null) {
+                exchangePayload.putContext(ExchangeContextKeys.EXCHANGE_EVENT_SOURCE, EventSource.WORKFLOW);
+            } else {
+                exchangePayload.putContext(ExchangeContextKeys.EXCHANGE_EVENT_SOURCE, EventSource.INTEGRATION);
+            }
+        }
+    }
+
+    private static void putContextIfNecessary(ExchangePayload exchangePayload, String key, Serializable value) {
+        if (exchangePayload.containsKey(key)) {
```
Comment (zakersimonjack) [inline core/context/src/main/java/com/milesight/beaveriot/context/util/ExchangeContextHelper.java:54]: Should be context here?

File: core/rule-engine/rule-engine-core/src/main/java/com/milesight/beaveriot/rule/observe/RuleEngineContextInterceptor.java [hunk 2]
```diff
@@ -40,6 +41,7 @@ protected void cacheOutputArguments(NamedNode definition, Exchange exchange) {
         if (definition.getId().startsWith(RuleFlowIdGenerator.FLOW_ID_PREFIX)) {
             String configNodeId = RuleFlowIdGenerator.removeNamespacedId(exchange.getFromRouteId(), definition.getId());
             exchange.setProperty(configNodeId, exchange.getIn().getBody());
+            exchange.setProperty(ExchangeHeaders.EXCHANGE_FLOW_ID, exchange.getFromRouteId());
```
Comment (zakersimonjack) [inline core/rule-engine/rule-engine-core/src/main/java/com/milesight/beaveriot/rule/observe/RuleEngineContextInterceptor.java:44]: I think this should be in `cacheFromArguments`

File: core/context/src/main/java/com/milesight/beaveriot/context/util/ExchangeContextHelper.java (hunk 1 above)
Comment (leonside) [reply core/context/src/main/java/com/milesight/beaveriot/context/util/ExchangeContextHelper.java]: fix

**Review by zakersimonjack — APPROVED** [review]

File: core/rule-engine/rule-engine-core/src/main/java/com/milesight/beaveriot/rule/observe/RuleEngineContextInterceptor.java (hunk 2 above)
Comment (leonside) [reply core/rule-engine/rule-engine-core/src/main/java/com/milesight/beaveriot/rule/observe/RuleEngineContextInterceptor.java]: fix

**Pull request merged**